unstoppable-ios-app/domains-manager-ios/Modules/EnterPasscode/DigitalKeyboard/DigitalKeyboardViewController.swift

```diff
@@ -34,12 +45,64 @@ final class DigitalKeyboardViewController: UIViewController {
 extension DigitalKeyboardViewController {
     func reset() {
         passcodeInputView.reset()
+        setupWarningLabel()
+    }
+    
+    func setupWarningLabel() {
+        guard let warningType = delegate?.getWarningType() else {
+            Debugger.printFailure("No delegate for keyboard found")
+            return
+        }
+        switch warningType {
+        case .none: resetWarningLabel()
+        case .wipe: setWipingLabel(String.Constants.allDataWillBeWiped)
+        case .lock: setWaitingLabel(String.Constants.appWillBeUnlocked)
+        }
+    }
+    
+    func resetWarningLabel() {
+        warningLabel.layer.masksToBounds = true
+        warningLabel.layer.cornerRadius = 8
+        warningLabel.text = ""
+        warningLabel.backgroundColor = .systemBackground
+        warningLabel.isHidden = true
```

Comment:
Is this setup necessary if at the end warning label will be hidden?

Reply:
Some of the parameters will be used in other cases, so this also serves as 'initializer'